nursereg/tests.py

```diff
@@ -766,31 +830,277 @@ def test_create_vumi_contact_passport(self):
+        # Execute
+        contact = tasks.update_create_vumi_contact.apply_async(
+            kwargs={"nursereg_id": nursereg.data["id"],
+                    "client": client})
+        result = contact.get()
+        # Check
+        self.assertEqual(result["msisdn"], "+27821237777")
+        self.assertEqual(result["groups"], [])
+        self.assertEqual(result["extra"], {
+            "nc_dob": "1951-01-02",
+            "nc_sa_id_no": "5101025009086",
+            "nc_is_registered": "true",
+            "nc_id_type": "sa_id",
+            "nc_faccode": "123456",
+            "nc_source_name": "Test Nurse Source",
+            "nc_persal": "11114444",
+            "nc_subscription_type": "11",
+            "nc_subscription_rate": "4",
+            "nc_subscription_seq_start": "1",
+        })
+        postsubs = Subscription.objects.all().count()
+        # check no additional subscription created
+        self.assertEqual(postsubs, presubs)
+
+    def test_create_vumi_contact_change_old_nr(self):
+        # Test mocks an external registration - no existing Vumi contact
+        # Setup
+        # make existing contact with msisdn 27821237777
+        self.make_existing_contact({
+            u"key": u"knownuuid",
+            u"msisdn": u"+27821237777",
+            u"user_account": u"knownaccount",
+            u"extra": {
+                "nc_dob": "1951-01-02",
+                "nc_sa_id_no": "5101025009086",
+                "nc_is_registered": "true",
+                "nc_id_type": "sa_id",
+                "nc_faccode": "123456",
+                "nc_source_name": "Test Nurse Source",
+                "nc_subscription_type": "11",
+                "nc_subscription_rate": "4",
+                "nc_subscription_seq_start": "11",
+            }
+        })
+        # nurse registration - change old nr
+        nursereg = self.make_nursereg(
+            post_data=TEST_REG_DATA["change_old_nr"])
+        client = self.make_client()
+        presubs = Subscription.objects.all().count()
+        # Execute
+        contact = tasks.update_create_vumi_contact.apply_async(
+            kwargs={"nursereg_id": nursereg.data["id"],
+                    "client": client})
+        result = contact.get()
+        # Check
+        self.assertEqual(result["msisdn"], "+27821234444")
+        self.assertEqual(result["groups"], [])
+        self.assertEqual(result["extra"], {
+            "nc_dob": "1951-01-02",
+            "nc_sa_id_no": "5101025009086",
+            "nc_is_registered": "true",
+            "nc_id_type": "sa_id",
+            "nc_faccode": "123456",
+            "nc_source_name": "Test Nurse Source",
+            "nc_subscription_type": "11",
+            "nc_subscription_rate": "4",
+            "nc_subscription_seq_start": "13",
+        })
+        postsubs = Subscription.objects.all().count()
+        # check two additional subscriptions created
+        self.assertEqual(postsubs, presubs+2)
```

Comment:
Surely there should only be one additional created? No-one should have two active nurse messagesets?